server/src/main/scala/actors/PartitionedPortStateActor.scala

```diff
@@ -0,0 +1,210 @@
+  }
+}
+
+trait PartitionedPortStateActorLike {
+  val now: () => SDateLike
+  val context: ActorContext
+  val journalType: StreamingJournalLike
+  val terminals: List[Terminal]
+
+  val queueUpdatesProps: (Terminal, SDateLike) => Props =
+    (terminal: Terminal, day: SDateLike) => {
+      Props(new TerminalDayQueuesUpdatesActor(day.getFullYear(), day.getMonth(), day.getDate(), terminal, now, journalType))
+    }
+
+  val staffUpdatesProps: (Terminal, SDateLike) => Props =
+    (terminal: Terminal, day: SDateLike) => {
+      Props(new TerminalDayStaffUpdatesActor(day.getFullYear(), day.getMonth(), day.getDate(), terminal, now, journalType))
+    }
+
+  val queueUpdatesSupervisor: ActorRef
+  val staffUpdatesSupervisor: ActorRef
+}
+
+trait ProdPartitionedPortStateActor extends PartitionedPortStateActorLike {
+  override val queueUpdatesSupervisor: ActorRef = context.system.actorOf(Props(new UpdatesSupervisor[CrunchMinute, TQM](now, terminals, queueUpdatesProps)))
+  override val staffUpdatesSupervisor: ActorRef = context.system.actorOf(Props(new UpdatesSupervisor[StaffMinute, TM](now, terminals, staffUpdatesProps)))
+}
+
+class PartitionedPortStateActor(flightsActor: ActorRef,
+                                queuesActor: ActorRef,
+                                staffActor: ActorRef,
+                                val now: () => SDateLike,
+                                val terminals: List[Terminal],
+                                val journalType: StreamingJournalLike) extends Actor with ProdPartitionedPortStateActor {
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  implicit val ec: ExecutionContextExecutor = context.dispatcher
+  implicit val mat: ActorMaterializer = ActorMaterializer.create(context)
+  implicit val timeout: Timeout = new Timeout(60 seconds)
+
+  def processMessage: Receive = {
+    case msg: SetCrunchQueueActor =>
+      log.info(s"Received crunchSourceActor")
+      flightsActor ! msg
+
+    case msg: SetSimulationActor =>
+      log.info(s"Received simulationSourceActor")
+      queuesActor ! msg
+
+    case StreamInitialized => sender() ! Ack
+
+    case StreamCompleted => log.info(s"Stream completed")
+
+    case StreamFailure(t) => log.error(s"Stream failed", t)
+
+    case flightsWithSplits: FlightsWithSplitsDiff =>
+      val replyTo = sender()
+      askThenAck(flightsWithSplits, replyTo, flightsActor)
+
+    case noUpdates: PortStateMinutes[_, _] if noUpdates.isEmpty =>
+      sender() ! Ack
+
+    case someQueueUpdates: PortStateQueueMinutes =>
+      val replyTo = sender()
+      askThenAck(someQueueUpdates.asContainer, replyTo, queuesActor)
+
+    case someStaffUpdates: PortStateStaffMinutes =>
+      val replyTo = sender()
+      askThenAck(someStaffUpdates.asContainer, replyTo, staffActor)
+
+    case GetState =>
+      log.warn("Ignoring GetState request (for entire state)")
+
+    case PointInTimeQuery(millis, request: GetStateForDateRange) =>
```

Comment:
we should look into breaking these functions out to make them more testable and composeable

Reply:
noted